### PR TITLE
Build the budget application using ROR

### PR DESCRIPTION
**Pull request by andersonlebon**
### Hello dear reviewer,
I want just to highlight all tasks that were performed in this PR 
- [ ] Set up the repository and tools.
- [ ] Create my models according to the ERD diagram. without forgetting validations and tests.
- [ ] Implement the authentication and authorization.
- [ ] Create the page to add a category.
- [ ] Create the home page.
- [ ] Create the page to add a transaction.
- [ ] Create the transactions page.
- [ ] Make sure my app is tested adequately.
- [ ] Deploy the project and test for final details.
- [ ] Record a video for my project.
- [ ] Create a good README and PR description

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,113 @@
...
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Microverse
+- Original design idea by Gregoire Vella on [Behance](https://www.behance.net/gregoirevella).
+
+## License
+
+[MIT](./LICENSE)
```
Comment (Dipeshtwis) [inline README.md:113]: - [ ] The Licence link is broken. Please fix that to make the README professional.

File: app/assets/stylesheets/categories.scss [hunk 2]
```diff
@@ -1,3 +1,73 @@
 // Place all the styles related to the categories controller here.
 // They will automatically be included in application.css.
 // You can use Sass (SCSS) here: https://sass-lang.com/
+
+.categories {
+  position: relative;
+}
+
+.scan {
+  // position: absolute;
```
Comment (Dipeshtwis) [inline app/assets/stylesheets/categories.scss]: - [ ] I would recommend removing the commented code to make the codebase cleaner.

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -1,3 +1,79 @@
...
+  form {
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    background-color: #fff;
+    border-top: 2px solid #dee2e6;
+    border-bottom: 2px solid #dee2e6;
+
+    .field {
+      //   border-top: 2px solid rgb(212, 212, 212);
```
Comment (Dipeshtwis) [inline app/assets/stylesheets/users.scss:49]: - [ ] Here as well. I would recommend removing the commented code to make the codebase cleaner.

File: app/controllers/transacs_controller.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  def create
+    @categories = current_user.categories
+    @transac = @categories.find(params[:category_id]).transacs.create(transac_params)
+    @transac.save
+    redirect_to user_category_path(current_user, params[:category_id])
+    # if @transac.save
+    #   redirect_to user_category_path(current_user, )
+    # else
+    #   render :new
+    # end
```
Comment (Dipeshtwis) [inline app/controllers/transacs_controller.rb:25]: - [ ] Here as well. I would recommend removing the commented code to make the codebase cleaner.

File: app/models/transac.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Transac < ApplicationRecord
+  belongs_to :category
+
+  validates :amount, presence: true
+  # has_many :categories, dependent: :destroy
```
Comment (Dipeshtwis) [inline app/models/transac.rb]: - [ ] Here as well. I would recommend removing the commented code to make the codebase cleaner.

File: app/models/user.rb [hunk 6]
```diff
@@ -1,10 +1,8 @@
...
-
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+         :recoverable, :validatable
   has_many :categories, dependent: :destroy
-  has_many :transactions, dependent: :destroy
+  # has_many :transacs, dependent: :destroy
```
Comment (Dipeshtwis) [inline app/models/user.rb]: - [ ] The user can have many transactions. Please fix the association between `users` and `transactions`.

File: README.md [hunk 7]
```diff
@@ -1,26 +1,113 @@
...
-* Database creation
+## Built With
 
-* Database initialization
+- Ruby on Rails
+- PostgreSQL
 
-* How to run the test suite
+## Video description
+[Loom video](https://loom.com/share/9fb0fa00b7c147ad928599dbcdf16334)
```
Comment (Dipeshtwis) [inline README.md]: - [ ] In the video presentation, please highlight a piece of the code you find interesting or are proud of making.

File: README.md [hunk 8]
```diff
@@ -1,26 +1,113 @@
...
 
-* Configuration
+A mobile web application where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what.
 
-* Database creation
+## Built With
 
-* Database initialization
+- Ruby on Rails
+- PostgreSQL
```
Comment (Dipeshtwis) [inline README.md:17]: - [ ] I would recommend deploying the project and it must be accessible online. I heard that you have already deployed 5 websites on Heroku, but I would suggest that if any of the projects is frontend please move it to netlify or if any application is not that important please delete and kindly deploy this application online. As this is a requirement please deploy it

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @andersonlebon, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 8 above)
Comment (andersonlebon) [reply README.md:17]: - [ ] Hello @Dipeshtwis  kindly all of the 5 projects deployed on Heroku are backEnds and are in use. I would have deleted them but they are in use. Kindly where else should I deploy my app??

Comment (andersonlebon) [reply README.md:17]: - [ ] I am trying to use GitHub for that bu is not working?

**Comment by andersonlebon** [comment]
# Hello dear reviewer
After a conversation with the previous reviewer on slack, I explained why the App is not working, and he told me to write a short note to the next review explaining why. 
In fact, the App is not deployed because the maximum of the app on Heroku is reached and all of them are in use as the backend for some reason.

**Comment by Davidosky007** [comment]
## Hi @andersonlebon 👋, 

## Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Uploading nice-job.gi
![thumbs-up-gif
f…]()


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**